src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -2072,10 +2072,71 @@ public boolean isJavaScriptEnabled() {
             return getOptions().isJavaScriptEnabled();
         }
 
+        /**
+         * Enables/disables JavaScript support.
+         * Short-hand method to ease discovery of feature + improve readability
+         * @see com.gargoylesoftware.htmlunit.WebClientOptions#setJavaScriptEnabled(boolean)
+         */
         public void setJavaScriptEnabled(boolean enabled) {
             getOptions().setJavaScriptEnabled(enabled);
         }
 
+        /**
+         * Enables/disables JavaScript support.
+         * Fluent method to ease discovery of feature + improve readability
+         * @see com.gargoylesoftware.htmlunit.WebClientOptions#setJavaScriptEnabled(boolean)
+         */
+        public WebClient withJavaScriptEnabled(boolean enabled) {
+            setJavaScriptEnabled(enabled);
+            return this;
+        }
+
+        public boolean isThrowExceptionOnFailingStatusCode() {
+            return getOptions().isThrowExceptionOnFailingStatusCode();
+        }
+
+        /**
+         * Changes the behavior of this webclient when a script error occurs.
+         * Short-hand method to ease discovery of feature + improve readability
+         * @see com.gargoylesoftware.htmlunit.WebClientOptions#setThrowExceptionOnFailingStatusCode(boolean)
+         */
+        public void setThrowExceptionOnFailingStatusCode(boolean enabled) {
+            getOptions().setThrowExceptionOnFailingStatusCode(enabled);
+        }
+
+        /**
+         * Changes the behavior of this webclient when a script error occurs.
+         * Fluent method to ease discovery of feature + improve readability
+         * @see com.gargoylesoftware.htmlunit.WebClientOptions#setThrowExceptionOnFailingStatusCode(boolean)
+         */
+        public WebClient withThrowExceptionOnFailingStatusCode(boolean enabled) {
+            setThrowExceptionOnFailingStatusCode(enabled);
+            return this;
+        }
+
+        public boolean isRedirectEnabled() {
```

Comment:
I don't understand why some of these have JavaDoc and not others.

Reply:
Simply because currently they are far less popular in the project (only one call of `isThrowExceptionOnFailingStatusCode`) compared to the one that I write JavaDoc for.

Now every added methods have their related JavaDoc, and also the `xxxJavaScriptEnabled` as they are similar.